assignment-client/src/entities/EntityServer.cpp

```diff
@@ -376,9 +376,7 @@ void EntityServer::nodeAdded(SharedNodePointer node) {
 
 void EntityServer::nodeKilled(SharedNodePointer node) {
     EntityTreePointer tree = std::static_pointer_cast<EntityTree>(_tree);
-    tree->withWriteLock([&] {
```

Comment:
I don't think this lock is necessary.  deleteDescendantsOfAvatar sometimes calls deleteEntitiesByID which already locks the tree